io-engine/src/rebuild/nexus_rebuild.rs

```diff
@@ -30,18 +37,26 @@ use super::{
 /// that there is no concurrent between the nexus and the rebuild.
 /// This is a frontend interface that communicates with a backend runner which
 /// is the one responsible for the read/writing of the data.
-pub struct NexusRebuildJob(RebuildJob);
+pub struct NexusRebuildJob {
+    job: RebuildJob,
+}
+
+pub struct NexusRebuildJobStarter {
```

Comment:
For Snapshot rebuild, do We need to have similar structure ? Can We have generic rebuild starter ?

Reply:
No, this is only needed for nexus because the rebuild map is added later, after we create the rebuild job.